modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/provider/PowwowServiceProviderFactory.java

```diff
@@ -14,46 +14,15 @@
 
 package com.liferay.powwow.provider;
 
-import com.liferay.powwow.provider.bbb.BBBPowwowServiceProvider;
 import com.liferay.powwow.provider.zoom.ZoomPowwowServiceProvider;
 
-import java.util.HashMap;
-import java.util.Map;
-
 /**
  * @author Marco Calderon
  */
 public class PowwowServiceProviderFactory {
 
-	public static PowwowServiceProvider getPowwowServiceProvider(
-		String providerType) {
-
-		if (!_powwowServiceProviders.containsKey(providerType)) {
-			throw new IllegalArgumentException("Invalid provider type");
-		}
-
-		return _powwowServiceProviders.get(providerType);
-	}
-
-	private static final Map<String, PowwowServiceProvider>
-		_powwowServiceProviders;
+	public static PowwowServiceProvider getPowwowServiceProvider() {
 
-	static {
-		_powwowServiceProviders = new HashMap<>();
-
-		PowwowServiceProvider bbbPowwowServiceProvider =
-			new BBBPowwowServiceProvider();
-
-		_powwowServiceProviders.put(
-			bbbPowwowServiceProvider.getPowwowServiceProviderKey(),
-			bbbPowwowServiceProvider);
-
-		PowwowServiceProvider zoomPowwowServiceProvider =
-			new ZoomPowwowServiceProvider();
-
-		_powwowServiceProviders.put(
-			zoomPowwowServiceProvider.getPowwowServiceProviderKey(),
-			zoomPowwowServiceProvider);
+		return new ZoomPowwowServiceProvider();
```

Comment:
Do not return a new instance every time. The original version (with Map), there is only one instance of each provider type. We can consider each of these:
- Have a static var to hold `ZoomPowwowServiceProvider`
- Or, keep the static map with one entry point to `ZoomPowwowServiceProvider`
- Or, maybe completely remove this factory class

Reply:
I Have a static var to hold ZoomPowwowServiceProvider